### ISSUE TITLE
[Bug] Copy-Paste works fine within CherryTree, but it doesn't work if its any other application.

### ISSUE THREAD

**Issue by PranavBhattarai** [giuspen/cherrytree#2200]
```
CherryTree 0.99.53
Fedora Workstation 37
Wayland, GNOME ver 43.2
```

Neither the shortkey (i.e. Ctrl + C) nor right-clicking any word to copy-and-paste works.

[Screencast from 2023-01-18 14-08-42.webm](https://user-images.githubusercontent.com/32763112/213121280-9f5e585b-fae3-431c-80c2-b61a9d375342.webm)

In the video above, I have tried to copy a text named "data" a couple of times and tried to paste it into a note editor but it doesn't work/respond. But if you try to copy-paste words inside CherryTree, it **does** work correctly. 

Edit: doesn't ---> does (typo)

**Comment by NoamHamnich** [comment]
Same problem.
Notice that when running CherryTree as root, copy paste works

**Comment by giuspen** [comment]
I have a virtual machine with Fedora 37 Gnome Wayland so I tried this but couldn't reproduce using the default text editor as recipient. Is this happening systematically for you or only some times?

**Comment by PranavBhattarai** [comment]
> Is this happening systematically for you or only sometimes?
No, a guy commented a day before yesterday. I got the email about his comments. But he/she deleted the comment for some reason, I guess. He/she said that he/she had to lunch the CherryTree with **root permission** (which is not a good approach).

This bug is so weird. I use ChatGPT (in Firefox) these days for documenting a lot of topics. And not being able to copy-paste the topics between two apps is really annoying/infuriating. 

If I copy something from CherryTree to Firefox, nothing is pasted.
If I copy something from Firefox to CherryTree, it works fine. 
Things should work vice-versa, but sadly its not the case.
I can't copy anything outside is CherryTree.

Is there a flag/command that I can use in order to trace the bug, then do let me know. But this bug exists for real.

And, no, this happens quite a lot of time. In previous releases, there was a rare experience. But nowadays, it's so frequent, I can say, "This happens **ALL**  the time".

**Comment by giuspen** [comment]
If instead of Copy (Ctrl+C) you use Copy as Plain Text (Shift+Ctrl+C) does it make any difference?

Would you be able to build cherrytree from the latest source code as described in https://github.com/giuspen/cherrytree#building-cherrytree-on-fedora and run from the terminal and report if you see differences? And also if you see any strange print on the terminal?

I did see the post later deleted. Can you also just for once and for sake of testing run as root cherrytree and see if that works?

**Comment by PranavBhattarai** [comment]
> If instead of Copy (Ctrl+C) you use Copy as Plain Text (Shift+Ctrl+C) does it make any difference?

I tried both the Copy-As-Plain-Text option and shortkey (Shift+Ctrl+C), but I don't think it made any difference.

### CherryTree launched from the terminal without root.
```
~ ❯❯❯ flatpak run net.giuspen.cherrytree
[2023-01-27 10:51:04.028] [gtk] [critical] Class::register_derived_type(): base_query.type_name is NULL.
[2023-01-27 10:51:04.033] [   ] [debug] /home/pranav/.var/app/net.giuspen.cherrytree/config/cherrytree/config.cfg parsed
[2023-01-27 10:51:04.130] [   ] [debug] autosave is started
[2023-01-27 10:51:04.368] [   ] [debug] Node name: Unit 10
[2023-01-27 10:51:33.231] [   ] [debug] shift images in MenuBar/context menu
[2023-01-27 10:52:04.216] [   ] [debug] autosave: no needs to save file
[2023-01-27 10:53:04.275] [   ] [debug] autosave: no needs to save file 
```

### CheeryTree launched from the terminal with root.
```
root@x ~# flatpak run net.giuspen.cherrytree
[2023-01-27 10:56:04.080] [gtk] [warning] Couldn't connect to accessibility bus: Failed to connect to socket /run/user/1000/at-spi/bus: No such file or directory
Gtk-Message: 10:56:04.112: Failed to load module "canberra-gtk-module"
Gtk-Message: 10:56:04.112: Failed to load module "pk-gtk-module"
Gtk-Message: 10:56:04.122: Failed to load module "canberra-gtk-module"
Gtk-Message: 10:56:04.122: Failed to load module "pk-gtk-module"
[2023-01-27 10:56:04.122] [gtk] [critical] Class::register_derived_type(): base_query.type_name is NULL.
[2023-01-27 10:56:04.129] [   ] [warning] /root/.var/app/net.giuspen.cherrytree/config/cherrytree/config.cfg missing
[2023-01-27 10:56:04.558] [   ] [debug] autosave is started
[2023-01-27 10:56:14.495] [   ] [debug] shift images in MenuBar/context menu
[2023-01-27 10:56:35.890] [   ] [debug] Node name: Routine
[2023-01-27 10:56:46.439] [   ] [debug] Node name: Unit 1
root@x ~#
```

> Can you also just for once and for sake of testing run as root cherrytree and see if that works?

@giuspen  yes, copy-paste works if I launch CherryTree as root.

**Comment by giuspen** [comment]
Looks like an issue with flatpak and wayland. I shall attach a native package for you to try instead.

**Comment by giuspen** [comment]
Can you try https://www.giuspen.net/software/cherrytree-0.99.53.99~Fedora37_amd64.rpm
It is marked as 0.99.53 but it is in fact the latest build, very very close to 0.99.54
If can be useful I'll release a package for Fedora37 as well in the 0.99.54 release that will happen on Sunday

**Comment by PranavBhattarai** [comment]
The domain is down or something? @giuspen 
I can't seem to visit the website, tried even google search to manually go to your official website.

**Comment by MJimitater** [comment]
@PranavBhattarai hmm strange.. I can access the website without problems (Im from Germany). Perhaps the problem is connected with the geo-location somehow.. could you re-route your geolocation?
BTW, nice to see that website is being maintained, just saw now new demo pics etc.

**Comment by PranavBhattarai** [comment]
@Mjimitater can you please upload the rpm package here? 

Re-routing my geolocation means paid VPN service. 
And my country doesn't normally allow/support Paypal AFAIK.

I tried Tor Browser to see if the website works since it uses "Tor Network", but the result is the same.
The website doesn't load.

**Comment by giuspen** [comment]
@PranavBhattarai I tried to upload the package here but without success. It allows me to upload such binaries only when I do the release ( https://github.com/giuspen/cherrytree/releases ). I plan to release tomorrow and I will create also a Fedora37 package is that ok? otherwise just send me an email privately and I will answer attaching the package. My email is in help--about dialog

**Comment by PranavBhattarai** [comment]
I downloaded the rpm package from my office network. But failed to install due to dependencies issues as you can see:
```
~/Downloads ❯❯❯ sudo rpm -i cherrytree-0.99.53.99_Fedora37_amd64.rpm                                                                ✘ 1
error: Failed dependencies:
	gtksourceviewmm3 is needed by cherrytree-0.99.53-1.x86_64
	libcurl-minimal is needed by cherrytree-0.99.53-1.x86_64
	libgtksourceviewmm-3.0.so.0()(64bit) is needed by cherrytree-0.99.53-1.x86_64
	libspdlog.so.1()(64bit) is needed by cherrytree-0.99.53-1.x86_64
	libxml++ is needed by cherrytree-0.99.53-1.x86_64
	libxml++-2.6.so.2()(64bit) is needed by cherrytree-0.99.53-1.x86_64
	spdlog is needed by cherrytree-0.99.53-1.x86_64
~/Downloads ❯❯❯                                
```
@giuspen Should I try to install all dependencies one by one?

**Comment by giuspen** [comment]
I'm surprised about the xorg libraries.
It is safe though to install packages from the standard library.

**Comment by PranavBhattarai** [comment]
@giuspen Sorry I didn't install the above package but **we have good news**.

The bug seems to be resolved/healed in the new version of **CherryTree 0.99.54**. I can now copy-paste between CT and other apps (like Firefox). So I guess we can close this issue?

**Comment by giuspen** [comment]
Thank you @PranavBhattarai is it the 0.99.54 flatpak version working or the fedora native package that I built?

**Comment by PranavBhattarai** [comment]
@giuspen Flatpak

**Comment by Nyanraltotlapun** [comment]
Can this be reopened or I better to create a new one?

```
Ubuntu 22.04
Linux 6.5.0-1025-oem
CherryTree 1.1.4 (snap)
Wayland Gnome and KDE
```

Cannot copy text from CherryTree to any other APP.
Can copy from others apps to CherryTree.

**Comment by DimShpInt** [comment]
Hi, I have the same problem but I can copy it into a text editor, browser and libreoffice. 
I can copy text to other programs if I run cherrytree in sudo.
